### DISCUSSION TITLE
what is autoshape

### DISCUSSION THREAD

**Discussion by chenzd01** [ultralytics/yolov5#4223] [Q&A]
What autoshape can be used for ？and in what scenarios we can use it？

**Comment by glenn-jocher** [comment]
@syxuming AutoShape is used for all YOLOv5 PyTorch Hub models to allow them to accept almost any input image type directly:
https://github.com/ultralytics/yolov5/blob/c2c958c350407b630bbbf063cefbd64cea7d8c81/models/common.py#L230-L239
